### PR TITLE
Fields and relationships refactoring.

### PR DESCRIPTION
**Pull request by mshima**
- Extract entities, fields and relationship preparation into functions.

Related to https://github.com/jhipster/generator-jhipster/issues/12178.

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/entity/index.js [hunk 1]
```diff
@@ -1204,6 +772,404 @@ class EntityGenerator extends BaseBlueprintGenerator {
...
+            entity.clientRootFolder ? `${entity.clientRootFolder}-${entity.entityStateName}` : entity.entityStateName
+        );
+
+        entity.reactiveRepositories = entity.reactive && ['mongodb', 'cassandra', 'couchbase', 'neo4j'].includes(entity.databaseType);
+
+        entity.differentTypes.push(entity.entityClass);
+        entity.i18nToLoad.push(entity.entityInstance);
+        entity.i18nKeyPrefix = `${entity.angularAppName}.${entity.entityTranslationKey}`;
+
+        entity.hasUserField = !!entity.relationships.find(relationship => this.isBuiltInUserEntity(relationship.otherEntityName));
```
Comment (murdos) [inline generators/entity/index.js:851]: See https://github.com/jhipster/generator-jhipster/pull/12176/files#diff-c026af5fb08143a94ab3665821943885L855 : this property is not used in template, not need to save it.
```suggestion
        let hasUserField = !!entity.relationships.find(relationship => this.isBuiltInUserEntity(relationship.otherEntityName));
```

File: generators/entity/index.js [hunk 2]
```diff
@@ -1204,6 +772,404 @@ class EntityGenerator extends BaseBlueprintGenerator {
...
+        entity.saveUserSnapshot =
+            entity.applicationType === 'microservice' &&
+            entity.authenticationType === 'oauth2' &&
+            entity.hasUserField &&
```
Comment (mshima) [inline generators/entity/index.js:855]:
```suggestion
            hasUserField &&
```

File: generators/entity/index.js [hunk 3]
```diff
@@ -35,6 +35,54 @@ const JHIPSTER_CONFIG_DIR = constants.JHIPSTER_CONFIG_DIR;
 
 const stringify = data => JSON.stringify(data, null, 4);
 
+const BASE_TEMPLATE_DATA = {
+    skipUiGrouping: false,
+    haveFieldWithJavadoc: false,
+    existingEnum: false,
+    searchEngine: false,
+
+    hasUserField: false,
```
Comment (mshima) [inline generators/entity/index.js:44]:
```suggestion
```

File: generators/entity/index.js [hunk 4]
```diff
@@ -1204,6 +771,404 @@ class EntityGenerator extends BaseBlueprintGenerator {
...
+            entity.clientRootFolder ? `${entity.clientRootFolder}-${entity.entityStateName}` : entity.entityStateName
+        );
+
+        entity.reactiveRepositories = entity.reactive && ['mongodb', 'cassandra', 'couchbase', 'neo4j'].includes(entity.databaseType);
+
+        entity.differentTypes.push(entity.entityClass);
+        entity.i18nToLoad.push(entity.entityInstance);
+        entity.i18nKeyPrefix = `${entity.angularAppName}.${entity.entityTranslationKey}`;
+
+        let hasUserField = !!entity.relationships.find(relationship => this.isBuiltInUserEntity(relationship.otherEntityName));
```
Comment (mshima) [inline generators/entity/index.js:850]:
```suggestion
        let hasUserField = entity.relationships.some(relationship => this.isBuiltInUserEntity(relationship.otherEntityName));
```

File: generators/entity/index.js [hunk 5]
```diff
@@ -1204,6 +771,401 @@ class EntityGenerator extends BaseBlueprintGenerator {
...
+            entity.clientRootFolder ? `${entity.clientRootFolder}-${entity.entityStateName}` : entity.entityStateName
+        );
+
+        entity.reactiveRepositories = entity.reactive && ['mongodb', 'cassandra', 'couchbase', 'neo4j'].includes(entity.databaseType);
+
+        entity.differentTypes.push(entity.entityClass);
+        entity.i18nToLoad.push(entity.entityInstance);
+        entity.i18nKeyPrefix = `${entity.angularAppName}.${entity.entityTranslationKey}`;
+
+        let hasUserField = entity.relationships.some(relationship => this.isBuiltInUserEntity(relationship.otherEntityName));
```
Comment (mshima) [inline generators/entity/index.js:850]:
```suggestion
        const hasUserField = entity.relationships.some(relationship => this.isBuiltInUserEntity(relationship.otherEntityName));
```

**Pull request merged**